### PR TITLE
Upgrade vulnerable deps: superagent & mocha

### PR DESCRIPTION
**Pull request by BigBlueHat**
Scary stuff, code. 😁 

Tests still pass, fwiw. 😉

### CODE REVIEW

**Comment by BigBlueHat** [comment]
I've also added the `package-lock.json` file [as recommended by npm](https://docs.npmjs.com/files/package-lock.json).

Cheers!
🎩

**Comment by tripu** [comment]
Tests on the main branch don't pass now (but not because of this PR).
Waiting to get #34 merged before reviewing this one.

File: package.json [hunk 1]
```diff
@@ -22,12 +22,12 @@
   },
   "dependencies": {
     "async": "2.1.4",
```
Comment (tripu) [inline package.json:24]:
```suggestion
    "async": "2.6.1",
```

File: package.json [hunk 2]
```diff
@@ -22,12 +22,12 @@
...
-    "superagent": "3.4.1"
+    "superagent": "^3.8.3"
```
Comment (tripu) [inline package.json:25]:
```suggestion
    "superagent": "4.0.0"
```

File: package.json [hunk 3]
```diff
@@ -22,12 +22,12 @@
...
   },
   "devDependencies": {
     "browserify": "14.0.0",
```
Comment (tripu) [inline package.json:28]:
```suggestion
    "browserify": "16.2.3",
```

File: package.json [hunk 4]
```diff
@@ -22,12 +22,12 @@
...
     "expect.js": "0.3",
-    "mocha": "3.2.0"
+    "mocha": "^5.2.0"
```
Comment (tripu) [inline package.json:30]:
```suggestion
    "mocha": "5.2.0"
```

**Review by tripu — CHANGES_REQUESTED** [review]
Thank you, @BigBlueHat!

* Do you mind rebasing against `origin/gh-pages`? I just fixed a recent error there, and tests on this branch won't pass now otherwise.
* We prefer exact versions; ie, no caret (see my suggestions).
* If we update all dependencies to their latest versions, all 3 vulns reported by `npm audit` are fixed (see my suggestions).
* I already confirmed that tests pass with these suggested changes, and that at least one dependent project ([Unitas](https://github.com/w3c/Unitas/)) works fine with these changes, too.

**Comment by tripu** [comment]
@BigBlueHat, I merged your PR as it was, and then applied my suggestions: https://github.com/w3c/node-w3capi/commit/6288d46955c79eb376c4a0fb0aa60d3e82fc3141
Those solve one vulnerability alert. The build is [clean now](https://github.com/w3c/node-w3capi/network/alerts).

**Pull request merged**